doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+# Popup
```

Comment:
Does vim `:help` works with markdown files? 🤔 (probably not as far as I know)...

If not, then we should better put these files close to the source code instead, for example: `lua/nui/popup/README.md`.

For components without separate directory, you can move them (e.g. move `lua/nui/input.lua` into `lua/nui/input/init.lua`) and add the `README.md` file in there.

Reply:
I don't think it works with markdown yup.

Hmm, I think it would be easier for the user to just pop into the `doc` folder and find all the docs there. 

This way you will have one readme per component and the user will have to go back and forth if they want to read it. True that there will be an option to visit all of them from the main readme but still I think them being all in one folder Is simpler. 

Plus it doesn't clutter up the source code with unnecessary files.

But if you prefer readmes in the directory directly I'll move them no problem.

Reply:
Well, keeping it with the source and naming them "README.md" has benefits actually.

From the user's perspective - they can option `lua/nui/popup` folder on GitHub and GitHub will render the `README.md` file automatically (without needing to open the markdown file separately). Without the name `README.md` it won't happen.

From the maintainer's perspective - it's easier to edit the markdown file in the same folder as the source code, instead of editing the markdown file from a different folder altogether.